### DISCUSSION TITLE
Meaning of a port "state" vs "connection"

### DISCUSSION THREAD

**Discussion by qm210** [djipco/webmidi#395] [Q&A]
Hey again,

I got another question that I couldn't figure out from the official docs. Sorry if I was just unable to find. I tried.

What seems to be the difference between port.connection (pending / open / closed) and port.state (connected / disconnected)?

In order to know "I can send notes to that output", does one check for both properties (connection == "open" && state == "connected") or is one of these sufficient?

best,
qm210

**Comment by djipco** [comment]
Both `port.connection` and `port.state` are properties of the [MIDIPort](https://www.w3.org/TR/webmidi/#idl-def-MIDIPort) object defined by the lower-level Web MIDI API. They are simply exposed by WEBMIDI.js for convenience.

The `port.connection` property represents the connection state of the device (`open`, `close` or `pending`). It is of type [MIDIPortConnectionState](https://www.w3.org/TR/webmidi/#idl-def-MIDIPortConnectionState). When WEBMIDI.js starts, all ports are automatically opened. So, you really shouldn't worry about opening (or closing) ports. You can listen to the `opened` event if you wish.

The `port.state` property represents the state of the device (`connected` or `disconnected`). It is of little use since a `disconnected` port will not even be accessible.
